tests/test_gradient.py

```diff
@@ -37,8 +37,10 @@ def test_gradientFWI(shape, kernel, space_order):
     wave = setup(shape=shape, spacing=spacing, dtype=np.float64,
                  kernel=kernel, space_order=space_order,
                  nbpml=10+space_order/2)
-    m0 = smooth10(wave.model.m.data, wave.model.shape_domain)
-    dm = np.float32(wave.model.m.data - m0)
+
+    m0 = Function(name='m0', grid=wave.model.m.grid, space_order=space_order)
+    m0.data[:] = smooth10(wave.model.m.data, wave.model.m.shape_domain)
+    dm = np.float32(wave.model.m.data - m0.data)
```

Comment:
If `dtype` is configurable, shouldn't you be using the `dtype` variable?

Reply:
maybe, just replicated ``master``. What do you suggest I change here, and into what? don't know why the write of this test explicitly set ``dm`` to ``np.float32``. I didn't want to alter this, just to be safe...

Reply:
There is a variable in scope here, called `dtype`, that contains the `dtype`.

Reply:
> There is a variable in scope here

i cannot see it